### ISSUE TITLE
Support 3D models in the datasets server and in the dataset viewer

### ISSUE THREAD

**Issue by severo** [huggingface/datasets-server#1003]
Note that the "file viewer" on the Hub already renders the 3D models (.glb) using https://modelviewer.dev/:

https://huggingface.co/datasets/allenai/objaverse/blob/main/glbs/000-000/000074a334c541878360457c672b6c2e.glb

https://user-images.githubusercontent.com/1676121/229077088-ef9bfde2-2e9a-4b12-8ac9-2d0ed3283df7.mov

related to #979

**Comment by severo** [comment]
maybe related: https://github.com/huggingface/datasets/issues/6388

**Comment by severo** [comment]
blocked by https://github.com/huggingface/datasets/issues/7195

**Comment by julien-c** [comment]
we can brainstorm about the UX maybe (i don't expect we should load all models on the page at once – IMO there should be a manual action from user to load + maybe load first couple of row by default) cc @gary149 @cfahlgren1

**Comment by cfahlgren1** [comment]
> we can brainstorm about the UX maybe (i don't expect we should load all models on the page at once – IMO there should be a manual action from user to load + maybe load first couple of row by default) cc @gary149 @cfahlgren1

I wonder if you could somehow show a preview (img) of the model in the dataset and when you click the img you can load the model in the modal similar to how we show expanded pictures. See lazy loaded model viewer [here](https://modelviewer.dev/examples/loading/#customizeLoad) too. 

![image](https://github.com/user-attachments/assets/26316d41-da18-4c13-9130-9464227f1365)

![image](https://github.com/user-attachments/assets/42484382-225d-43f8-92b4-5e45bb7b24ef)

**Comment by gary149** [comment]
> I wonder if you could somehow show a preview (img) of the model in the dataset and when you click the img you can load the model in the modal similar to how we show expanded pictures. See lazy loaded model viewer [here](https://modelviewer.dev/examples/loading/#customizeLoad) too.

Sure, that would work. How would you obtain an image for each 3D model and each dataset, though? It's seems quite complex. Maybe a a simple placeholder each time would be simpler.

<img width="465" alt="image" src="https://github.com/user-attachments/assets/60126251-2a00-4891-a549-c8e5d9c083f9">

**Comment by cfahlgren1** [comment]
yeah that part is tricky, given there are only a few datasets on the hub probably not worth the effort. 
I like the icon approach. Logo is 🔥

**Comment by julien-c** [comment]
what about my idea of loading a few (can be 1) first items
